test/functional/signature/cert-chain.bats

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bats
+
+load "../testlib"
+
+test_setup() {
+
+	create_test_environment "$TEST_NAME"
+	create_bundle -n test-bundle -f /test-file "$TEST_NAME"
+
+	key_generation_cmd=$(cat <<-EOM
+			# Create a root key and cert
+			openssl req -x509 -sha512 -days 1 -newkey rsa:4096  -keyout "$TEST_NAME/root.key" -out "$TEST_NAME/root.pem" -nodes -subj '/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=localhost'
+			# Create a intermediate key and cert signed by root
+			openssl req -sha512 -nodes -newkey rsa:4096 -keyout "$TEST_NAME/intermediate.key" -out "$TEST_NAME/intermediate.csr" -subj '/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=intermediate'
+			openssl x509 -days 1 -req -in "$TEST_NAME/intermediate.csr" -CA "$TEST_NAME/root.pem" -CAkey "$TEST_NAME/root.key" -set_serial 0x1 -out "$TEST_NAME/intermediate.pem"
+			# Create a self signed certificate for intermediate
+			openssl req -days 1 -new -x509 -key "$TEST_NAME/intermediate.key" -out "$TEST_NAME/intermediate_self_signed.pem" -set_serial 0x1 -subj '/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=intermediate'
+			# Sign the MoM with intermediate cert
+			openssl smime -sign -binary -in "$WEBDIR"/10/Manifest.MoM -signer "$TEST_NAME/intermediate.pem" -inkey "$TEST_NAME/intermediate.key" -out "$WEBDIR"/10/Manifest.MoM.sig -outform DER
+			# Sign the MoM with self signed intermediate cert
+			openssl smime -sign -binary -in "$WEBDIR"/10/Manifest.MoM -signer "$TEST_NAME/intermediate_self_signed.pem" -inkey "$TEST_NAME/intermediate.key" -out "$WEBDIR"/10/Manifest.MoM.sig_self_signed -outform DER
+			# Create a certificate store with both root and self signed intermediate
+			cat "$TEST_NAME/root.pem" > "$TEST_NAME/trust.pem"
+			cat "$TEST_NAME/intermediate_self_signed.pem" >> "$TEST_NAME/trust.pem"
+	EOM
+	)
+
+	run sudo sh -c "$key_generation_cmd"
+	assert_status_is 0
+
```

Comment:
I don't think you need to use the `run` command to execute those commands, as far as I understand run needs to be used to execute the command you want to test cause it will capture the output and will prevent the script from failing if the command being run fails (which sometimes is expected for tests). Test prerequisites like these can just be executed as in any other bash script, and if the command fails it will fail the whole test script. Which is expected. That way you would not have to use the assertion there (again since that is supposed to be used with the thing you want to test).

Reply:
I was unsure about that too. I'll rake the runs out. I prefer that too.